dj/materialization/jobs/materialization_job.py

```diff
@@ -0,0 +1,69 @@
+"""
+Available materialization jobs.
+"""
+import abc
+from typing import Optional
+
+from dj.models.engine import Dialect
+from dj.models.node import MaterializationConfig
+from dj.service_clients import QueryServiceClient
+
+
+class MaterializationJob(abc.ABC):  # pylint: disable=too-few-public-methods
+    """
+    Base class for a materialization job
+    """
+
+    dialect: Optional[Dialect] = None
+
+    def __init__(self):
+        ...
+
+    @abc.abstractmethod
+    def execute(
```

Comment:
Would `schedule` be a better name for this method? It sends the instructions to the query service but may not exactly execute it until the next scheduled instance runs right?

Reply:
+1, see the 1st word in the docstring.

Reply:
Makes sense! I renamed it to schedule.